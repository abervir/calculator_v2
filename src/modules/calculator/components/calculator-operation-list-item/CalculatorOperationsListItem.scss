.operation-label {
  display: flex;
  align-items: center;

  .icon {
    margin-right: 5px;
  }
}
