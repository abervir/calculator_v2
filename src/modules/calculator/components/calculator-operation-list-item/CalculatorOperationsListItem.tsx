import React from 'react';

import './CalculatorOperationsListItem.scss';

interface Props {
  title: string;
  icon: string;
}

export const CalculatorOperationsListItem = (props: Props) => {
  const { title, icon } = props;

  return (
    <div className='operation-label'>
      <i title={title} className={`icon ${icon}`}></i>
      <span>{title}</span>
    </div>
  );
};
