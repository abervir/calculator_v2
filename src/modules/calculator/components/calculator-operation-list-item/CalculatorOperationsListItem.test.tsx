import { shallow } from 'enzyme';
import React from 'react';

import { CalculatorOperationsListItem } from './CalculatorOperationsListItem';

describe('CalculatorOperationsListItem', () => {
  it('should render component', () => {
    const title = 'title';
    const icon = 'icon';

    expect(
      shallow(<CalculatorOperationsListItem title={title} icon={icon} />)
    ).toMatchSnapshot();
  });
});
