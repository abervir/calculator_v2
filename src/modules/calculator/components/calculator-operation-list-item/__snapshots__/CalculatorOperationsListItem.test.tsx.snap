// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalculatorOperationsListItem should render component 1`] = `
<div
  className="operation-label"
>
  <i
    className="icon icon"
    title="title"
  />
  <span>
    title
  </span>
</div>
`;
